packages/fast-components-react-base/src/viewport-positioner/viewport-positioner.tsx

```diff
@@ -0,0 +1,1011 @@
+            this.viewportRect.bottom - (this.anchorTop + this.anchorHeight);
+
+        switch (positionOption) {
+            case ViewportPositionerVerticalPositionLabel.top:
+                return spaceAbove;
+            case ViewportPositionerVerticalPositionLabel.insetTop:
+                return spaceAbove + this.anchorHeight;
+            case ViewportPositionerVerticalPositionLabel.insetBottom:
+                return spaceBelow + this.anchorHeight;
+            case ViewportPositionerVerticalPositionLabel.bottom:
+                return spaceBelow;
+        }
+    };
+
+    /**
+     *  Handle scroll events
+     */
+    private handleScroll = (): void => {
+        this.requestFrame();
+    };
+
+    /**
+     *  Handle anchor resize events
+     */
+    private handleAnchorResize = (entries: ResizeObserverEntry[]): void => {
+        const entry: ResizeObserverEntry = entries[0];
+        this.anchorHeight = entry.contentRect.height;
+        this.anchorWidth = entry.contentRect.width;
+
+        if (
+            this.state.currentVerticalPosition ===
+            ViewportPositionerVerticalPositionLabel.top
+        ) {
+            this.anchorBottom = this.anchorTop + this.anchorHeight;
+        } else {
+            this.anchorTop = this.anchorBottom - this.anchorHeight;
+        }
+
+        if (
+            this.state.currentHorizontalPosition ===
+            ViewportPositionerHorizontalPositionLabel.left
+        ) {
+            this.anchorRight = this.anchorLeft + this.anchorWidth;
+        } else {
+            this.anchorLeft = this.anchorRight - this.anchorWidth;
+        }
+
+        this.requestFrame();
+    };
+
+    /**
+     *  Handle collisions
+     */
+    private handleCollision = (
+        entries: IntersectionObserverEntry[],
+        observer: IntersectionObserver
+    ): void => {
+        entries.forEach((entry: IntersectionObserverEntry) => {
+            if (entry.target === this.rootElement.current) {
+                this.processPositionerCollision(entry);
+            } else {
+                this.processAnchorCollision(entry);
+            }
+        });
+
+        this.scrollTop = this.getViewportElement().scrollTop;
+
+        this.scrollLeft = this.getViewportElement().scrollLeft;
+
+        this.requestFrame();
+    };
+
+    /**
+     *  Update data based on anchor collisions
+     */
+    private processAnchorCollision = (anchorEntry: IntersectionObserverEntry): void => {
+        this.viewportRect = anchorEntry.rootBounds;
+        this.anchorTop = anchorEntry.boundingClientRect.top;
+        this.anchorRight = anchorEntry.boundingClientRect.right;
+        this.anchorBottom = anchorEntry.boundingClientRect.bottom;
+        this.anchorHeight = anchorEntry.boundingClientRect.height;
+        this.anchorWidth = anchorEntry.boundingClientRect.width;
+    };
+
+    /**
+     *  Update data based on positioner collisions
+     */
+    private processPositionerCollision = (
```

Comment:
"setPositionerCollision"

Reply:
Hmm...don't like "set" for this, how about "handlePositionerCollision"?